tests/checks/functions.fish

```diff
@@ -171,3 +175,17 @@ functions --no-details --details t
 # CHECKERR: ^
 # CHECKERR: (Type 'help functions' for related documentation)
 # XXX FIXME ^ caret should point at --no-details --details
+
+function term1 --on-signal TERM
+end
+function term2 --on-signal TERM
+end
+function term3 --on-signal TERM
+end
+
+functions --handlers-type signal
+# CHECK: Event signal
+# CHECK: SIGTRAP fish_sigtrap_handler
```

Comment:
This used to print a second "Event signal" after the SIGTRAP

Reply:
> fix an earlier regression introduced in the Rust code

Yes, the issue was new in the rust version, caused by enum comparisons being deep.

Probably something to look out for - if we replace C++ enums with Rust enums, the C++ version only compares variants.

>2. there is a behavior difference in the sorting, which is probably fine, but should perhaps be mentioned in the release notes?

I do not believe this is a thing that is expected to be stable. This *isn't* sorted alphabetically, it's sorted *by definition order* - and used to be in reverse definition (oldest last) order.